vendor/src/github.com/docker/libnetwork/sandbox_store.go

```diff
@@ -132,7 +134,16 @@ func (sb *sandbox) storeUpdate() error {
 		sbs.Eps = append(sbs.Eps, eps)
 	}
 
-	return sb.controller.updateToStore(sbs)
+	err := sb.controller.updateToStore(sbs)
+	if err == datastore.ErrKeyModified {
+		// When we get ErrKeyModified it is sufficient to just
+		// go back and retry.  No need to get the object from
+		// the store because we always regenerate the store
+		// state from in memory sandbox state
+		goto retry
```

Comment:
@mrjana the question is more, can `err == datastore.ErrKeyModified` happen multiple times in the same goroutine, resulting in multiple retries, and if so, can it happen indefinitely in some case?